### DISCUSSION TITLE
Follow the all-contributors specification

### DISCUSSION THREAD

**Discussion by benjagm** [json-schema-org/community#390] [General]
JSON Schema is a heterogeneous project compound by multiple repositories inside and outside the Github Organization what makes difficult to maintain an updated list of the project contributors. In addition, each project is using a different flavour of contributing guidelines what in the end create friction in the experience of new and existing contributors.

## Proposal

To reduce friction and provide a consistent way to praise highlight and praise contributors I suggest to start using [All Contributors specification](https://allcontributors.org/).

**What is All Contributors specification**

All Contributors is a specification for recognizing contributors to an open-source project in a way that rewards every contribution, not just code. The basic idea is to use the project README (or another prominent public documentation page in the project) to recognize the contributions of members of the project community.

More information about the spec: https://allcontributors.org/docs/en/overview

Examples of projects using it: https://allcontributors.org/docs/en/project/implementations

My intend with this discussion is to gather feedback and identify any other alternatives.
Please, vote YES if you approve the proposal or NO if you disagree. 

Thank you! 🙂

**Comment by MikeRalphson** [comment]
Datapoints: [AsyncAPI](https://github.com/asyncapi/spec/blob/master/README.md#contributors) follows All-Contributors model / process. I raised a [PR](https://github.com/OAI/OpenAPI-Specification/pull/2627) to adopt it for OpenAPI but we ultimately decided not to go down this route. One 'objection' from a fellow TSC member was that it looked like giving everyone a particpatory medal, or was an attempt a gamifying the process of contributing to the specification, and it was questioned whether this added real value beyond the normal GitHub [contributors](https://github.com/OAI/OpenAPI-Specification/graphs/contributors) graph/insight panel or indeed helped to encourage any more meaningful contributions. So in some senses All-Contributors is a solution in search of a problem.

I will abstain on voting either way.

**Reply by Julian** [reply]
I'm abstaining as well for essentially the same reasons, I'm aware of all-contributors but I honestly don't know what problem it solves. If someone wants it though I'm not strongly against it.

**Comment by benjagm** [comment]
One benefit is allowing to add non-code contributors, like mentors or volunteers in events.

**Comment by Relequestual** [comment]
I would previously have objected to this, however I'm now more in favour.

One problem I think it solves is, as a potential new contributor, who do I find out someone to talk to about a non-code or non-spec related contribution? Having a list of those who have contributed in a specific area could be helpful. Granted, it doesn't signify how much the contribution was.

> One 'objection' from a fellow TSC member was that it looked like giving everyone a particpatory medal, or was an attempt a gamifying the process of contributing to the specification, and it was questioned whether this added real value beyond the normal GitHub [contributors](https://github.com/OAI/OpenAPI-Specification/graphs/contributors) graph/insight panel or indeed helped to encourage any more meaningful contributions

I can totally understand and share some of those concerns. I am of mixed opinions on "participatory medals". In some sense, any commit appears on the contributors graph, which is not that disimilar to participatory medals. My first PR was fixing some typo in the jQuery docs. It was small, but everyone has to start somewhere. If a participatory medal could lead to a "gold", then regardless of how many do nothing further, and considering it's minimal effort once set up (as I understand), it feels like a it has the potential to be a win even if not immediatly.

**Reply by MikeRalphson** [reply]
Datapoint: my first PR to OAS was a missing full stop.

**Comment by Relequestual** [comment]
Interestingly, GitHub found it too much effort and removed it from their docs repo: https://github.com/github/docs/pull/576

**Reply by benjagm** [reply]
> too much effort

If the effort is because the huge amount of contributions I'd love that to be our future problem. 😃

**Comment by benjagm** [comment]
Something really interesting four our use case is this method will allow us to recognize Code contributions in repos outside our org like Bowtie.

**Comment by gregsdennis** [comment]
Someone in the GH Maintainers group posted a GH Action they made to collect contributions: https://github.com/JoshuaKGoldberg/all-contributors-auto-action.  I've been meaning to have a look at it for `json-everything`, but I haven't gotten around to it yet.

**Reply by gregsdennis** [reply]
I had a look at the spec and bot today.

The spec appears fine, but it seems that the bot has been largely abandoned.  There are a number of open issues, and there hasn't been an update for a couple years.

I decided not to use this in my repo and instead simply rely on GitHub's "contributors" section which seems to pick up a lot.
